src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -777,6 +777,58 @@ object FunctionRegistry {
     "alts" -> "Array of alternate allele sequences."
   )
 
+  val combineVariantsStruct = TStruct(Array(("variant", TVariant, "Resulting combined variant."),
+    ("laIndices", TDict(TInt, TInt), "Mapping from new to old allele index for the left variant."),
+    ("raIndices", TDict(TInt, TInt), "Mapping from new to old allele index for the right variant.")
+    ).zipWithIndex.map { case ((n, t, d), i) => Field(n, t, i, Map(("desc", d))) })
+
+  registerAnn("combineVariants",
+    combineVariantsStruct, { (left: Variant, right: Variant) =>
+      if (left.contig != right.contig || left.start != right.start)
+        fatal(s"Only variants with the same contig and position can be combined. Left was $left, right was $right.")
+
+      val (longer, shorter, swapped) = if (left.ref.length > right.ref.length) (left, right, false) else (right, left, true)
+      val ref_diff = longer.ref.substring(shorter.ref.length)
+
+      if (longer.ref.substring(0, shorter.ref.length) != shorter.ref)
+        fatal(s"Variants ref bases mismatch in combineVariants. Left ref: ${ left.ref }, right ref: ${ right.ref }")
+
+      val short_alleles_index = mutable.Map[Int, Int](0 -> 0)
+      val short_alleles = new mutable.ArrayBuffer[AltAllele](initialSize = shorter.nAltAlleles)
+
+      Range(0, shorter.nAltAlleles).foreach({
+        i =>
+          val alt = shorter.altAlleles(i).alt + ref_diff
+          val new_index = longer.altAlleles.indexWhere(_.alt == alt)
```

Comment:
This is quadratic in complexity, would prefer to make a map from longer alt to index, then use `get` and match no that.